### PR TITLE
[ftxui] update to 2.0.0

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
- Switch to version-semver
- Add version 2.0.0. Released 14 days ago:
https://github.com/ArthurSonzogni/FTXUI/releases/tag/v2.0.0

**Describe the pull request**

- #### What does your PR fix?  
  - Add version 2.0.0
  - Use `version-semver` instead of `version-date`

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

  Similar to https://github.com/microsoft/vcpkg/pull/22293, uwp is not supported. I haven't touched `ci.baseline.txt`. Should I? Not clear to me.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/ftxui/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ftxui",
-  "version-date": "2021-10-17",
-  "port-version": 1,
+  "version-semver": "2.0.0",
+  "port-version": 2,
```
Comment (JackBoosY) [inline ports/ftxui/vcpkg.json:4]:
```suggestion
```
Since we updated the version, we don't need to update port-version field.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ftxui/vcpkg.json (hunk 1 above)
Comment (ArthurSonzogni) [reply ports/ftxui/vcpkg.json]: Thanks! This makes a lot of sense.
Done.

File: ports/ftxui/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ftxui",
-  "version-date": "2021-10-17",
+  "version-semver": "2.0.0",
```
Comment (JackBoosY) [inline ports/ftxui/vcpkg.json:3]: Note to other reviewer: See https://github.com/ArthurSonzogni/FTXUI/discussions/207.

**Review by JackBoosY — APPROVED** [review]

File: ports/ftxui/vcpkg.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
   "port-version": 1,
```
Comment (ras0219-msft) [inline ports/ftxui/vcpkg.json:4]: Conventionally we remove the port-version (resetting it to 0) whenever the main version updates. However, this isn't a significant enough issue to warrant blocking the PR.

**Review by ras0219-msft — APPROVED** [review]
LGTM, thanks for the PR!

**Pull request merged**